package.yaml

```diff
@@ -39,6 +39,7 @@ default-extensions:
   - NamedFieldPuns
```

Comment:
Continuing the discussion from [here](https://github.com/serokell/xrefcheck/pull/212#discussion_r1013042345)

(This topic is not 100% strictly related the problem this PR aims to solve, but it is related to `blockListF` and `indentF`. So I think this should be solved in a separate commit in this PR, or in a separate PR altogether)


`indentF` adds a newline at the end (unless the text to be indented already ends in a newline):

```hs
$ indentF 2 "a"
"  a\n"

$ indentF 2 "a\n"
"  a\n"
```

Same thing with `blockListF`:
```
$ blockListF ["a","b"]
"- a\n- b\n"

$ blockListF ["a\n","b\n"]
"- a\n- b\n"
```

One could argue that, in general, adding a `\n` at the end is good.

However, I would argue this makes these functions unsuitable to be used in combination with string interpolation.

Consider this case:
```
λ> :{ 
[int||
aaa
#{indentF 2 "bbb"}
ccc
|]
:}
```

I would have expected the result to be

```
aaa
  bbb
ccc
```

But in reality it's:

```
aaa
  bbb

ccc
```

At the moment, to work around this, we're currently adding a `\` after `indentF` / `blockListF`. The backslash gets rid of one of the `\n` characters.

```hs
λ> :{ 
[int||
aaa
#{indentF 2 "bbb"}\
ccc
|]
:}
```

```
aaa
  bbb
ccc
```

But this is error-prone, it's very easy to forget to add a `\`.

I think we should have versions of `indentF` and `blockListF` that do not append a trailing newline, which would make them suitable to be used with string interpolation. Suggestion: we could call them `interpolateIndentF` and `interpolateBlockList`.